### ISSUE TITLE
Drivers for Braswell not working on Synology DS 216+ (Sonoff Dongle Plus)

### ISSUE THREAD

**Issue by mariusrga** [robertklep/dsm7-usb-serial-drivers#57]
I've tried to use the drivers for my Synology NAS (Braswell), but they do not work. Getting below error:

insmod: ERROR: could not insert module /lib/modules/cp210x.ko: Invalid module format

Appreciate if you could help me with this.

Thank you.

**Comment by robertklep** [comment]
Hi! You didn't download the correct file(s), please follow the instructions [here](https://github.com/robertklep/dsm7-usb-serial-drivers#downloading-a-module).

**Comment by mariusrga** [comment]
Mostly yes.
1. Downloaded the files from here: https://github.com/robertklep/dsm7-usb-serial-drivers/tree/main/modules/braswell
2. Copied all the files to /lib/modules
3. tried to use insmod to insert them (ex: sudo insmod /lib/modules/cp210x.ko), but was getting those errors for each file.
4. usb-serial-drivers.sh does not exist in the Braswell folder, so not sure what is it.
Thank you.

**Comment by robertklep** [comment]
Mostly, but not entirely, because you didn't download the actual `.ko` files, otherwise you wouldn't get that particular error.

**Comment by mariusrga** [comment]
My platform: Linux HomeNAS 3.10.108 #42962 SMP Tue Oct 18 15:05:36 CST 2022 x86_64 GNU/Linux synology_braswell_216+

**Comment by mariusrga** [comment]
> Mostly, but not entirely, because you didn't download the actual `.ko` files, otherwise you wouldn't get that particular error. The file `usb-serial-drivers.sh` isn't in the Braswell folder, it's in the [top-level folder](https://github.com/robertklep/dsm7-usb-serial-drivers).

These are the files I downloaded, for Braswell. If these are not correct, then, which are please? https://github.com/robertklep/dsm7-usb-serial-drivers/tree/main/modules/braswell/dsm-7.1

**Comment by robertklep** [comment]
Click (_don't_ download) each `.ko` file and press the Download button, as explained in the download instructions that I linked to.

![image](https://user-images.githubusercontent.com/1190916/213713968-5ecb47d3-b824-46d1-b7cd-becd2dab143d.png)

**Comment by mariusrga** [comment]
I used sudo wget then the link to each file, for ex;
sudo wget https://github.com/robertklep/dsm7-usb-serial-drivers/blob/main/modules/braswell/dsm-7.1/ch341.ko

Is this not OK?

**Comment by robertklep** [comment]
No, that's not okay, the correct link is https://github.com/robertklep/dsm7-usb-serial-drivers/raw/main/modules/braswell/dsm-7.1/ch341.ko
